lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -388,6 +394,33 @@ def configure(conf)
       @resource.labels.freeze
       @common_labels.freeze
 
+      # If prometheus monitoring is enabled, register metrics in the default
+      # registry and store metric objects for future use. If metrics are
+      # registed already, just store the metric objects.
+      if @prometheus_monitoring_enabled
+        require 'prometheus/client'
+
+        prometheus = Prometheus::Client.registry
+
+        # Exception-driven behavior to avoid synchronization errors
+        begin
+          @prometheus_http_requests_count = prometheus.counter(
```

Comment:
I didn't want to re-implement the "default solution", b/c there's actually pretty much synchronization code in the Prometheus library and copying it would result in a lot of code.

I don't see much profit in having several monitoring solution used at the same time. What do you think about the following: I create a base Monitoring class & a Prometheus implementation and you will create a custom implementation independently; then I change the `prometheus_monitoring_enabled` flag to `monitoring_type` with the string type, that can be one of the following: `none`, `prometheus` and later `json`?